File: solutions/524.longest-word-in-dictionary-through-deleting/longest-word-in-dictionary-through-deleting.py
class Solution:
    def findLongestWord(self, s, d):
        """
        :type s: str
        :type d: List[str]
        :rtype: str
        """
        def isSub(word):
            it = iter(s)
            return all(c in it for c in word)
        d.sort(key= lambda x : (-len(x), x))
        return next(filter(isSub, d), '')

File: solutions/524.longest-word-in-dictionary-through-deleting/longest-word-in-dictionary-through-deleting.java
class Solution {
    public String findLongestWord(String s, List<String> d) {
        d.sort(Comparator.comparing((String x) -> -x.length()).thenComparing((x)->(x)));
        for (String word : d) {
            int idx = 0;
            for (char c : s.toCharArray()) {
                if (idx >= word.length()) break;
                if (c == word.charAt(idx)) {
                    idx++;
                }
            }
            if (idx == word.length()) return word;
        }
        
        return "";
    }

}